### PR TITLE
Jetty 12 : XmlAppendable use Charset, not String

### PR DESCRIPTION
**Pull request by joakime**
Change to have XmlAppendable use Charset not String to create outputstream and whatnot.

Taken from PR #8597

### CODE REVIEW

File: jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlAppendable.java [hunk 1]
```diff
@@ -30,31 +33,31 @@
...
+        this(new OutputStreamWriter(out, charset), charset);
     }
 
     public XmlAppendable(Appendable out) throws IOException
     {
         this(out, 2);
     }
 
-    public XmlAppendable(Appendable out, String encoding) throws IOException
+    public XmlAppendable(Appendable out, Charset charset) throws IOException
```
Comment (joakime) [inline jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlAppendable.java]: This constructor is not used outside of this class, can probably be removed.

File: jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlAppendable.java [hunk 2]
```diff
@@ -30,31 +33,31 @@
...
     {
-        this(out, 2, encoding);
+        this(out, 2, charset);
     }
 
     public XmlAppendable(Appendable out, int indent) throws IOException
```
Comment (joakime) [inline jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlAppendable.java]: This constructor is not used outside of this class, can probably be removed.

File: jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlAppendable.java [hunk 3]
```diff
@@ -30,31 +33,31 @@
...
     {
-        this(out, indent, "utf-8");
+        this(out, indent, StandardCharsets.UTF_8);
     }
 
-    public XmlAppendable(Appendable out, int indent, String encoding) throws IOException
+    public XmlAppendable(Appendable out, int indent, Charset charset) throws IOException
```
Comment (joakime) [inline jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlAppendable.java]: This constructor is not used outside of this class, can probably be removed/collapsed into the constructors that are used.

File: jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlAppendable.java [hunk 4]
```diff
@@ -30,31 +33,31 @@
     private final Stack<String> _tags = new Stack<>();
     private String _space = "";
 
-    public XmlAppendable(OutputStream out, String encoding) throws IOException
+    public XmlAppendable(OutputStream out, Charset charset) throws IOException
```
Comment (joakime) [inline jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlAppendable.java]: This constructor is never used with a Charset that isn't UTF-8, should probably just remove this constructor too and have only 1 constructor that has only 1 parameter `OutputStream`.

**Review by joakime — COMMENTED** [review]
I'm going to collapse these constructors into 1

File: jetty-ee10/jetty-ee10-quickstart/src/main/java/org/eclipse/jetty/ee10/quickstart/QuickStartGeneratorConfiguration.java [hunk 5]
```diff
@@ -16,6 +16,7 @@
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
```
Comment (sbordet) [inline jetty-ee10/jetty-ee10-quickstart/src/main/java/org/eclipse/jetty/ee10/quickstart/QuickStartGeneratorConfiguration.java]: Unnecessary import?

File: jetty-ee9/jetty-ee9-quickstart/src/main/java/org/eclipse/jetty/ee9/quickstart/QuickStartGeneratorConfiguration.java [hunk 6]
```diff
@@ -16,6 +16,7 @@
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
```
Comment (sbordet) [inline jetty-ee9/jetty-ee9-quickstart/src/main/java/org/eclipse/jetty/ee9/quickstart/QuickStartGeneratorConfiguration.java]: Unnecessary import?

**Review by sbordet — APPROVED** [review]
LGTM apart the possibly 2 unnecessary imports.

**Review by gregw — COMMENTED** [review]
retrospective +1

**Pull request merged**